src/renderers/WebGLRenderer.js

```diff
@@ -1010,380 +1010,393 @@ THREE.WebGLRenderer = function ( parameters ) {
 
-				var position = geometry.attributes[ 'position' ];
+			}
 
-				// render non-indexed triangles
+			var position = geometry.attributes[ 'position' ];
 
-				if ( geometry instanceof THREE.InstancedBufferGeometry && geometry.maxInstancedCount > 0 ) {
+			// render non-indexed triangles
 
-					var extension = extensions.get( 'ANGLE_instanced_arrays' );
+			if ( geometry instanceof THREE.InstancedBufferGeometry && geometry.maxInstancedCount > 0 ) {
 
-					if ( extension === null ) {
+				var extension = extensions.get( 'ANGLE_instanced_arrays' );
 
-						THREE.error( 'THREE.WebGLRenderer.setupVertexAttributes: using THREE.InstancedBufferGeometry but hardware does not support extension ANGLE_instanced_arrays.' );
-						return;
+				if ( extension === null ) {
 
-					}
+					THREE.error( 'THREE.WebGLRenderer.setupVertexAttributes: using THREE.InstancedBufferGeometry but hardware does not support extension ANGLE_instanced_arrays.' );
+					return;
 
-					if ( position instanceof THREE.InterleavedBufferAttribute ) {
+				}
 
-						extension.drawArraysInstancedANGLE( mode, 0, position.data.array.length / position.data.stride, geometry.maxInstancedCount ); // Draw the instanced meshes
+				if ( position instanceof THREE.InterleavedBufferAttribute ) {
 
-					} else {
+					extension.drawArraysInstancedANGLE( mode, 0, position.data.array.length / position.data.stride, geometry.maxInstancedCount ); // Draw the instanced meshes
 
-						extension.drawArraysInstancedANGLE( mode, 0, position.array.length / position.itemSize, geometry.maxInstancedCount ); // Draw the instanced meshes
+				} else {
 
-					}
+					extension.drawArraysInstancedANGLE( mode, 0, position.array.length / position.itemSize, geometry.maxInstancedCount ); // Draw the instanced meshes
 
-				} else {
+				}
 
-					if ( position instanceof THREE.InterleavedBufferAttribute ) {
+			} else {
 
-						_gl.drawArrays( mode, 0, position.data.array.length / position.data.stride );
+				if ( position instanceof THREE.InterleavedBufferAttribute ) {
 
-					} else {
+					_gl.drawArrays( mode, 0, position.data.array.length / position.data.stride );
 
-						_gl.drawArrays( mode, 0, position.array.length / position.itemSize );
+				} else {
 
-					}
+					_gl.drawArrays( mode, 0, position.array.length / position.itemSize );
 
 				}
 
-				_this.info.render.calls ++;
-				_this.info.render.vertices += position.array.length / position.itemSize;
-				_this.info.render.faces += position.array.length / ( 3 * position.itemSize );
-
 			}
 
-		} else if ( object instanceof THREE.PointCloud ) {
+			_this.info.render.calls ++;
+			_this.info.render.vertices += position.array.length / position.itemSize;
+			_this.info.render.faces += position.array.length / ( 3 * position.itemSize );
 
-			// render particles
+		}
 
-			var mode = _gl.POINTS;
+	}
 
-			var index = geometry.attributes.index;
+	function renderPointCloudDirect ( camera, lights, fog, material, geometry, object ) {
```

Comment:
I think the parameters here don't match the other methods...? Have you tested this code?